react-renderer/src/inspector/views/Mod.tsx

```diff
@@ -0,0 +1,161 @@
+		  // console.log("properties, shapeType", properties, shapeType, properties.w, properties.h);
+
+                  let [w, h] = [0, 0];
+
+                  if (shapeType === "Circle") {
+		      [w, h] = 
+			  [
+                              (properties.r.contents as number) * 2,
+                              (properties.r.contents as number) * 2,
+			  ];
+		  } else if (shapeType === "Square") {
+		      [w, h] = 
+			  [
+			      (properties.side.contents as number),
+			      (properties.side.contents as number),
+			  ];
+      } else if (shapeType === "Ellipse") {
+          [w, h] = 
+          [
+            (properties.rx.contents as number) * 2,
+            (properties.ry.contents as number) * 2,
+          ];
+      } else if (shapeType === "Arrow" || shapeType === "Line") {
+		      const [sx, sy, ex, ey] = [properties.startX.contents as number, 
+						properties.startY.contents as number, 
+						properties.endX.contents as number, 
+						properties.endY.contents as number];
+
+		      const padding = 50; // Because arrow may be horizontal or vertical, and we don't want the size to be zero in that case
+		      // size of bbox of arrow
+		      [w, h] = [Math.max(Math.abs(ex - sx), padding), Math.max(Math.abs(ey - sy), padding)];
+      } else if (shapeType === "Curve") {
+          [w, h] = [20, 20] // todo find a better measure for this... check with max?
+      }
+      else {
+		      [w, h] = [properties.w.contents as number, properties.h.contents as number];
+		  }
+
+                  return (
+                  <ShapeItem
+                    key={`shapePreview-${key}`}
+                    selected={selectedShape === key}
+                    onClick={() => this.setSelectedShape(key)}
+                  >
+                    <div>
+                      <svg viewBox={`0 0 ${w} ${h}`} width="50" height="50">
+                        {React.createElement(staticMap[shapeType], {
+                          shape: {
+                            ...properties,
+                            x: { tag: "FloatV", contents: 0 },
+                            y: { tag: "FloatV", contents: 0 },
+                          },
+                          canvasSize: [w, h],
+                        })}
+                      </svg>
+                    </div>
+                    <div style={{ margin: "0.5em" }}>
+                      <span>{properties.name.contents}</span>
+                    </div>
+                  </ShapeItem>
+                );
+              }
+            )}
+          </ul>
+        </div>
+        <div
+          style={{
+            // BUG: scroll doesnt really work
+            padding: "1em 1em 1em 1em",
+            overflow: "auto",
+            height: "100%",
+            flexBasis: 0,
+            flexGrow: 1,
+            boxSizing: "border-box"
+          }}
+        >
+          {/* todo enable only on last frame */}
```

Comment:
hm?

Reply:
Expired comment... I'll get it rid of it.

Reply:
Addressed in 2b9315a4a27f35fb751ac317f15cdbf9a18ec5e4